include/xtensor/xoperation.hpp

```diff
@@ -81,6 +81,43 @@ namespace xt
         }                                                              \
     }
 
+// This functor avoids implicit conversions of small integral types to 'int'
+// by returning the same type instead of 'auto'.
+#define UNARY_BITWISE_OPERATOR_FUNCTOR(NAME, OP)                   \
+    struct NAME                                                    \
+    {                                                              \
+        template <class A1>                                        \
```

Comment:
Question: is the name `A1` something that was used in similar bits of codes. If not, I would find e.g. `T` more logical (or even `A` or `E` which is used in many places)